### PR TITLE
Fix: Include "this" jinja object in dbt singular tests

### PR DESCRIPTION
**Pull request by crericha**

### CODE REVIEW

**Review by tobymao — APPROVED** [review]

File: sqlmesh/dbt/manifest.py [hunk 1]
```diff
@@ -172,13 +172,15 @@ def _load_tests(self) -> None:
             )
 
             test_model = _test_model(node)
+            node_config = _node_base_config(node)
+            node_config["database"] = node.database or self.target.database
```
Comment (izeigerman) [inline sqlmesh/dbt/manifest.py:176]: why do we do this?

Comment (crericha) [reply sqlmesh/dbt/manifest.py:176]: For singular tests, dbt returns the test database in the "this" obj. This is to match it.

Comment (izeigerman) [reply sqlmesh/dbt/manifest.py:176]: So the problem with this is that we no longer know whether the `database` was actually overridden for any given model.

Comment (crericha) [reply sqlmesh/dbt/manifest.py:176]: This is only for a `test` node.

**Review by izeigerman — APPROVED** [review]

**Pull request merged**